src/Implementation/Shared/Core/AsyncPump.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+// http://blogs.msdn.com/b/pfxteam/archive/2012/01/20/10259049.aspx
+
+namespace Microsoft.Threading
+{
```

Comment:
I left it exactly as was in the original blog post, but I can modify to our namespace if you think that would be better.